core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/Reloadable.scala

```diff
@@ -0,0 +1,248 @@
+package com.github.fit51.reactiveconfig.reloadable
+
+import cats.MonadError
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.typesafe.scalalogging.StrictLogging
+import monix.eval.Task
+import monix.eval.TaskLift
+import monix.eval.TaskLike
+import monix.execution.Scheduler
+import monix.reactive.Consumer
+import monix.reactive.Observable
+import monix.reactive.Observable.Operator
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import scala.util.Failure
+
+object Reloadable {
+  def apply[F[_], A](
+      initial: A,
+      ob: Observable[A]
+  )(implicit scheduler: Scheduler, F: MonadError[F, Throwable], T: TaskLike[F], L: TaskLift[F]): F[Reloadable[F, A]] =
+    (for {
+      connectableObservable <- Task.pure(ob.publish)
+      canceler   = connectableObservable.connect()
+      reloadable = new Reloadable[F, A](initial, connectableObservable)
+      fiber <- connectableObservable
+        .consumeWith(Consumer.foreach { newValue =>
+          reloadable.value = newValue
+        })
+        .start
+    } yield {
+      reloadable.canceler = (fiber.cancel >> Task.delay(canceler.cancel())).to[F]
+      reloadable
+    }).to[F]
+
+  type Id[X] = X
+
+  implicit val taskLikeId = new TaskLike[Id] {
+    override def apply[A](fa: Id[A]): Task[A] = Task.pure(fa)
+  }
+}
+
+/**
+  * Reloadable is a wrapped [[B]] value, that can be accessed at any time.
+  * Inside it encapsulates logic for updating [[B]] on [[A]] changes.
+  *
+  * @see [[ReloadableInternal]]
+  * @param initial   value of type [[B]]
+  * @param ob        observable of [[A]] changes
+  * @param start     function to get [[B]] from [[A]]. Is used in [[behaviour]]
+  * @param behaviour [[ReloadBehaviour]]
+  * @tparam F [_] reloading effect
+  * @tparam A input
+  * @tparam B output
+  **/
+class Reloadable[F[_], A] private (initial: A, ob: Observable[A])(
+    implicit scheduler: Scheduler,
+    F: MonadError[F, Throwable],
+    T: TaskLike[F],
+    L: TaskLift[F]
+) extends StrictLogging {
+  import Reloadable._
+
+  @volatile
+  private var value = initial
+
+  protected[reloadable] val observable: Observable[A] = ob
+
+  def unsafeGet: A =
+    value
+
+  def get: F[A] =
+    value.pure[F]
+
+  def forEachF(f: A => F[Unit]): F[Unit] =
```

Comment:
We decided to keep this method. It can be used to subscribe with some sideeffect function on config changes.
Will add some examples and tests with it.

Reply:
I added description for method and one test.